src/Symfony/Component/Routing/Matcher/ExpressionLanguageProvider.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Matcher;
+
+use Symfony\Component\ExpressionLanguage\ExpressionFunction;
+use Symfony\Component\ExpressionLanguage\ExpressionFunctionProviderInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Defines some ExpressionLanguage functions.
+ *
+ * @author Ahmed TAILOULOUTE <ahmed.tailouloute@gmail.com>
+ */
+class ExpressionLanguageProvider implements ExpressionFunctionProviderInterface
+{
+    /** @var RequestContext */
+    private $context;
+
+    public function __construct(RouterInterface $router)
+    {
+        $this->context = $router->getContext();
```

Comment:
`getContext()` is not part of `RouterInterface` - but we don't need this, do we? Isn't the context passed to the runtime function already?

Instead, we should pass the list of exposed functions here also, and remove the hardcoded `'env'` string in the implementation.

Reply:
we do need the context for the `evaluator` of the expression language (the 2nd callable) 

I removed the hardcoded `env`, now we can pass multiple functions to the `ExpressionLanguageProvider`

Reply:
I changed also `RouterInterface` with `RequestContextAwareInterface` (which is the interface that defines `getContext` and implemented by the router)